### PR TITLE
Improves Report Generator stability and fixes various bugs

### LINKED ISSUE #4730: Add New Crisis Event In Report
#### Expected Behavior
Have updated crisis events

#### Actual Behavior
The last event is `NewNormal`.

#### Potential Solution
Rename `NewNormal` and add more events. Suggestion from user:
[Requesting updates to Report generator - new crisis events and sync between QC and local versions
](https://www.quantconnect.com/forum/discussion/9257/requesting-updates-to-report-generator-new-crisis-events-and-sync-between-qc-and-local-versions/p1)

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by gsalaz98**
#### Description

  * Adds unit tests covering changes
  * Adds COVID-19 crisis plots
  * Adjusts styling of crisis plots for more pleasant viewing
  * Fixes bug where null BacktestResult caused application to crash
  * Order JSON bug fixes and stability improvements
  * MaxDrawdownReportElement now produces results for Live
  * Replaced Estimated Capacity w/ Days Live
  * Added Live marker to sharpe ratio
  * Added support for MOO and MOC orders in PortfolioLooper

#### Related Issue

Closes #4730 
#### Motivation and Context

Better reports
#### Requires Documentation Change

No
#### How Has This Been Tested?

Report generation on all platforms
#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Report/PortfolioLooper/PortfolioLooper.cs [hunk 1]
```diff
@@ -323,13 +324,24 @@ private IEnumerable<PointInTimePortfolio> ProcessOrders(IEnumerable<Order> order
...
+                    lastFillTime = order.Time;
+                }
+                else if (order.LastFillTime == null)
                 {
                     Log.Trace($"Order with ID: {order.Id} has been skipped because of null LastFillTime");
                     continue;
                 }
+                else
+                {
+                    lastFillTime = order.Time;
```
Comment (Martin-Molinero) [inline Report/PortfolioLooper/PortfolioLooper.cs:341]: Seems this else should be using `LastFillTime`? Also would be nice to see more tests for the `PortfolioLooper`

File: Report/ReportElements/DaysLiveReportElement.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+using System;
+using System.Linq;
+using QuantConnect.Packets;
+
+namespace QuantConnect.Report.ReportElements
+{
+    internal class DaysLiveReportElement : ReportElement
+    {
+        private LiveResult _live;
+        private BacktestResult _backtest;
```
Comment (Martin-Molinero) [inline Report/ReportElements/DaysLiveReportElement.cs:25]: `_backtest` isn't being used, it doesn't seems to be a required argument for `DaysLiveReportElement`?

File: Tests/Common/Orders/OrderJsonConverterTests.cs [hunk 3]
```diff
@@ -263,6 +263,123 @@ public void DeserializesNullLastFillTimeAndLastUpdateTime()
             TestOrderType(actual2);
         }
 
+        [Test]
+        public void DeserializesStringStatusAndNullTime()
```
Comment (Martin-Molinero) [inline Tests/Common/Orders/OrderJsonConverterTests.cs:267]: Nice 👍

File: Report/ReportElements/MaxDrawdownReportElement.cs [hunk 4]
```diff
@@ -42,7 +45,24 @@ public MaxDrawdownReportElement(string name, string key, BacktestResult backtest
...
+            if (_live == null)
+            {
+                return _backtest?.TotalPerformance?.PortfolioStatistics?.Drawdown.ToString("P1") ?? "-";
+            }
+
+            var backtestEquityPoints = new Series<DateTime, double>(ResultsUtil.EquityPoints(_backtest));
+            var liveEquityPoints = new Series<DateTime, double>(ResultsUtil.EquityPoints(_live));
+
+            var backtestDrawdownGroups = new DrawdownCollection(backtestEquityPoints, 5);
+            var liveDrawdownGroups = new DrawdownCollection(liveEquityPoints, 5);
```
Comment (Martin-Molinero) [inline Report/ReportElements/MaxDrawdownReportElement.cs:57]: Wonder why fixed `5` periods? Think it would be nice to see some tests for this guy too

**Review by Martin-Molinero — COMMENTED** [review]
Cool! 👍
In general I think we are missing some more unit tests tests

**Review by jaredbroad — COMMENTED** [review]
🥳

File: Tests/Report/DrawdownCollectionTests.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                new KeyValuePair<DateTime, double>(new DateTime(2020, 1, 2), 90000),
+                new KeyValuePair<DateTime, double>(new DateTime(2020, 1, 3), 100000),
+                new KeyValuePair<DateTime, double>(new DateTime(2020, 1, 4), 100000),
+                new KeyValuePair<DateTime, double>(new DateTime(2020, 1, 5), 80000)
+            });
+
+            var collection = DrawdownCollection.GetDrawdownPeriods(series, 1);
+
+            Assert.GreaterOrEqual(1.9999, collection.First().Drawdown);
+            Assert.LessOrEqual(collection.First().Drawdown, 2.0001);
```
Comment (Martin-Molinero) [inline Tests/Report/DrawdownCollectionTests.cs:42]: For `double` you can specify a delta for `Assert.AreEquals()`, suggest we also assert that `collection` enumerates a single value?

File: Tests/Report/PortfolioLooperTests.cs [hunk 6]
```diff
@@ -47,5 +47,71 @@ public void EmptyEquitySeriesDoesNotCrash()
...
+                (decimal)stopPrice,
+                (decimal)limitPrice,
+                new DateTime(2019, 1, 3, 5, 0, 5),
+                string.Empty
+            ));
+            var exitOrder = Order.CreateOrder(new SubmitOrderRequest(
+                orderType,
+                SecurityType.Equity,
+                Symbols.SPY,
+                1,
```
Comment (Martin-Molinero) [inline Tests/Report/PortfolioLooperTests.cs:80]: quantity should be negative for exitOrder?

**Review by Martin-Molinero — APPROVED** [review]
Thanks for the tests 👍, looks good just minor test suggestions

**Pull request merged**